indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -659,9 +663,14 @@
         _canActOn: function(category) {
             var self = this;
             var ids = self.options.actionOn.categories.ids;
-            return !(self.options.actionOn.categoriesWithSubcategories.disabled && category.deep_category_count) &&
-                   !(self.options.actionOn.categoriesWithEvents.disabled && category.has_events) &&
-                   !(self.options.actionOn.categories.disabled && _.contains(ids, category.id));
+            var categoriesWithSubcategories = self.options.actionOn.categoriesWithSubcategories;
+            var categoriesWithEvents = self.options.actionOn.categoriesWithEvents;
+
+            return !(categoriesWithSubcategories.disabled && category.deep_category_count) &&
+                   !(categoriesWithEvents.disabled && category.has_events) &&
+                   !(self.options.actionOn.categories.disabled && _.contains(ids, category.id)) ||
+                   (categoriesWithEvents.disabled && category.has_events &&                 // Make sure hubrid categories are treated
+                    !categoriesWithSubcategories.disabled && category.deep_category_count); // as categories with subcategories
```

Comment:
I think this covers all possible cases:
```js
var hasOnlyEvents = (category.has_events && !category.deep_category_count);
var hasSubcategories = !!category.deep_category_count;

if ((categoriesWithSubcategories.disabled && hasSubcategories) ||
    (categoriesWithEvents.disabled && hasOnlyEvents) ||
    (self.options.actionOn.categories.disabled && _.contains(ids, category.id))) {
    return false;
}
return true;
```

* If the category has subcategories and the categories with subcategories are not allowed, then immediately `return false`
* If the category has **only** events and the categories with events are not allowed, then `return false`. If it **doesn't** contain only events though, it means it contains subcategories as well (hybrid category). In that case it will follow the `return true` after the `if` block.

What do you guys think, @ThiefMaster , @OmeGak ?